### PR TITLE
Refact: Circle CI config

### LINKED ISSUE #5: Circle CI configuration
I was looking at the circle ci configuration and it needs some love, wanna assign this to someone @kewitz?

Thanks

### PR DESCRIPTION
**Pull request by rodrigo4244**
Fixes #5

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -49,12 +67,16 @@ jobs:
 
 workflows:
   version: 2
-  build-and-deploy:
+  test:
```
Comment (gtaltemann) [inline .circleci/config.yml]: Shouldn't this be broken down to `test` and `deploy`?

**Comment by gtaltemann** [comment]
Besides the note that I left for review, everything else looks right 💯

File: .circleci/config.yml [hunk 2]
```diff
@@ -49,12 +67,16 @@ jobs:
...
+          requires:
+            - checkout
       - deploy:
           requires:
-            - build
+            - test
+            - checkout
           filters:
             branches:
               only: master
```
Comment (jshrt) [inline .circleci/config.yml]: Nice, but I do not see any build process, doesn't this project need to be linted, built and then test and then and only then deployed?

File: .circleci/config.yml (hunk 1 above)
Comment (rodrigo4244) [reply .circleci/config.yml]: I don't think we need two workflows. We can deploy after doing tests on `master`.

Comment (rodrigo4244) [reply .circleci/config.yml]: But yeah, maybe we should have a standard for this.

File: .circleci/config.yml (hunk 2 above)
Comment (rodrigo4244) [reply .circleci/config.yml]: The command `yarn test` lints and test (I think mocha already builds for testing .ts files). From `package.json`:
```json
{
    "pretest": "yarn lint",
    "test": "nyc mocha \"test/**/*.spec.ts\" --exit",
    "posttest": "assert-changelog-update && assert-version-bump"
}
```

File: .circleci/config.yml (hunk 1 above)
Comment (rodrigo4244) [reply .circleci/config.yml]: All other projects are using only `test` workflow. Maybe we should keep as is because of consistency for now

**Comment by kewitz** [comment]
@rodrigo4244 please wrap this up and :shipit:.

**Comment by gtaltemann** [comment]
@rodrigo4244 what's the status on this?

**Pull request merged**